### PR TITLE
Add squash-commit-{title,message}

### LINKED ISSUE #147: Allow custom commit title/message when merge mode is "squash"
When using `squash` as merge mode, the merge commit title is always the PR name with the PR number.
It would be great to have the possibility to customize this, for example with a new input for the action.
Similarly, it would be good if we could customize the commit message which is currently always blank.

### PR DESCRIPTION
**Pull request by markelliot**
Fixes #147: Enable use of `squash-commit-{title,message}` to customize squash commit.

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -127,3 +127,38 @@ export async function pullRequestsForWorkflowRun(
...
+  squashTitle: boolean,
+  squashCommitTitle: string | null,
+  squashCommitMessage: string | null,
+  pullRequest: PullRequest,
+): {title: string | undefined, message: string | undefined} {
+  if (!isSquashCommit) {
+    return {title: undefined, message: undefined}
+  }
+
+  if (squashTitle) {
```
Comment (reitermarkus) [inline src/helpers.ts:142]: I think we should remove this. Either way this is a breaking change given the new inputs have default values.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Comment by reitermarkus** [comment]
Thanks, @markelliot, I left some comments.

File: src/input.ts [hunk 2]
```diff
@@ -28,6 +28,8 @@ export class Input {
   token: string
   mergeMethod: MergeMethod
   squashTitle: boolean
+  squashCommitTitle: string | null
+  squashCommitMessage: string | null
```
[the remark this answers is not in the document]
Comment (markelliot) [reply src/input.ts:32]: ah I wasn't sure -- it looks like they can't be null based on the types

File: src/helpers.ts (hunk 1 above)
Comment (reitermarkus) [reply src/helpers.ts:142]: Or maybe remove the defaults, since these pretty much mirror the GitHub defaults anyways.

File: README.md [hunk 3]
```diff
@@ -22,20 +22,20 @@ Ensure the following is set up in your repository settings before enabling this
...
-| `squash-title` | no | Use the pull request title as the commit message when squashing. |
-| `do-not-merge-labels` | no | When any of the labels in this comma-separated list is applied to a pull request, it will not be merged automatically. |
-| `required-labels` | no | Comma-separated list of labels that are required to be applied to a pull request for it to be merged automatically. |
-| `pull-request` | no | Try merging the specified pull request automatically. For example, you can pass an input from a `workflow_dispatch` event. |
-| `pull-request-author-associations` | no | Comma-separated list of required author associations for the pull request author. (By default, pull requests by any author are allowed.)|
-| `dry-run` | no | If set to `true`, will not actually merge pull requests but still perform all other checks. |
-
+| Name                               | Required | Description                                                                                                                                                                                                       |
```
Comment (markelliot) [inline README.md:27]: sorry about the reformat here, I can make the change more concise but VSCode ended up reformatting the entire table by default and I figured not too much harm.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: src/input.ts [hunk 4]
```diff
@@ -51,7 +52,8 @@ export class Input {
       }
     }
 
-    this.squashTitle = core.getInput('squash-title') === 'true'
+    this.squashCommitTitle = core.getInput('squash-commit-title') || undefined
+    this.squashCommitMessage = core.getInput('squash-commit-message') || undefined
```
[the remark this answers is not in the document]
Comment (markelliot) [reply src/input.ts:59]: done

File: src/input.ts [hunk 5]
```diff
@@ -53,6 +55,9 @@ export class Input {
 
     this.squashTitle = core.getInput('squash-title') === 'true'
```
Comment (reitermarkus) [inline src/input.ts:56]: This should now set the `squashCommitTitle` and `this.squashCommitMessage ` templates, if `true`.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -22,20 +22,20 @@ Ensure the following is set up in your repository settings before enabling this
...
-| `dry-run` | no | If set to `true`, will not actually merge pull requests but still perform all other checks. |
-
+| Name                               | Required | Description                                                                                                                                    |
+| ---------------------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| `token`                            | yes      | A GitHub Token other than the default `GITHUB_TOKEN` needs to be specified in order to be able to enable auto-merge.                           |
+| `merge-method`                     | no       | Specify which merge method to use. By default, will select the first one available in this order: `merge`, `squash`, `rebase`                  |
+| `squash-title`                     | no       | Use the pull request title as the commit message when squashing.                                                                               |
```
Comment (reitermarkus) [inline README.md:31]: Maybe add a note to this saying it is deprecated.

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
Thanks, @markelliot!

**Pull request merged**